match/lib/match/storage/google_cloud_storage.rb

```diff
@@ -81,6 +99,32 @@ def initialize(type: nil,
         ensure_bucket_is_selected
       end
 
+      def currently_used_team_id
+        if self.readonly
+          # In readonly mode, we still want to see if the user provided a team_id
+          # see `prefixed_working_directory` comments for more details
+          return self.team_id
+        else
+          spaceship = SpaceshipEnsure.new(self.username, self.team_id, self.team_name)
+          return spaceship.team_id
+        end
+      end
+
+      def prefixed_working_directory
+        # We fall back to "*", which means certificates and profiles
+        # from all teams that use this bucket would be installed. This is not ideal, but
+        # unless the user provides a `team_id`, we can't know which one to use
+        # This only happens if `readonly` is activated, and no `team_id` was provided
+        @_folder_prefix ||= currently_used_team_id
+        if @_folder_prefix.nil?
+          # We use a `@_folder_prefix` variable, to keep state between multiple calls of this
+          # method, as the value won't change. This way the warning is only printed once
+          UI.important("Looks like you run `match` in `readonly` mode, and didn't provide a `team_id`. This will still work, however it is recommended to provide a `team_id` in your Appfile or Matchfile")
+          @_folder_prefix = "*"
```

Comment:
So, if there is no team_id, we will prefix the paths with `*` character? What is the reason for this character? It feels over-loaded with bash directory globbing. Could we use something else that doesn't have incidental meaning?

Reply:
I'm actually not sure 😕  I just moved this code over but it would probalby be good to understand this cause the `*` does seem a bit dangerous if that is what its used for